### PR TITLE
BBD-599 Range input controls for navigation

### PR DESCRIPTION
**Pull request by Guo-Haowei**

### CODE REVIEW

File: src/core/action-creator.ts [hunk 1]
```diff
@@ -38,15 +38,31 @@ export function createActions(flux: FluxCapacitor) {
...
-        action(Actions.UPDATE_SEARCH, { ...<any>search, query: search.query && search.query.trim() }, {
-          ...metadata,
-          validator: {
-            payload: {
-              func: ({ query }) => !!query || query === null,
-              msg: 'search term is empty'
-            }
-          }
-        }),
+      action(Actions.UPDATE_SEARCH, 'query' in search ? { ...<any>search, query: search.query && search.query.trim() } : search, {
```
Comment (effervescentia) [inline src/core/action-creator.ts]: looks like the formatting is a little off here

File: test/unit/core/action-creator.ts [hunk 2]
```diff
@@ -126,6 +137,29 @@ suite('ActionCreator', ({ expect, spy, stub }) => {
       });
     });
 
+    describe('addRefinement()', () => {
+      it('should update a value refinement when two parameters are provided', () => {
+	const updateSearch = actions.updateSearch = spy();
```
Comment (effervescentia) [inline test/unit/core/action-creator.ts]: formatting

Comment (zydc2010) [inline test/unit/core/action-creator.ts]: indentation

File: src/core/action-creator.ts [hunk 3]
```diff
@@ -38,16 +38,32 @@ export function createActions(flux: FluxCapacitor) {
...
             payload: {
-              func: ({ query }) => !!query || query === null,
+              func: ({ query }) => !('query' in search) || (!!query || query === null),
               msg: 'search term is empty'
             }
           }
         }),
 
+      addRefinement: (field: string, valueOrLow: any, high: any = null) => {
+	if (typeof high === 'number') {
```
Comment (effervescentia) [inline src/core/action-creator.ts]: formatting

**Pull request merged**